app/controllers/report_controller/reports/editor.rb

```diff
@@ -1552,6 +1546,18 @@ def set_form_vars
     end
   end
 
+  def categories_hash
+    cats = Classification.categories.select { |c| c.show }
+    cats.delete_if { |c| c.entries.length == 0 }  # Omit categories for which entries dropdown would be empty.
+    cats.each_with_object({}) { |c, h| h[c.name] = c.description }
+  end
+
+  def entries_hash(category_name)
+    cat = Classification.find_by_name(category_name)
```

Comment:
Thanks @cben.  Maybe it's worth seeing how hard it is to do.  There are an unknown number of callers of this method so it would depend how much work it is to do and verify.